synapse/config/saml2_config.py

```diff
@@ -33,15 +52,18 @@ def read_config(self, config, **kwargs):
 
         self.saml2_enabled = True
 
-        import saml2.config
-
-        self.saml2_sp_config = saml2.config.SPConfig()
-        self.saml2_sp_config.load(self._default_saml_config_dict())
-        self.saml2_sp_config.load(saml2_config.get("sp_config", {}))
+        saml2_config_dict = self._default_saml_config_dict()
+        _dict_merge(saml2_config.get("sp_config", {}), saml2_config_dict)
 
         config_path = saml2_config.get("config_path", None)
         if config_path is not None:
-            self.saml2_sp_config.load_file(config_path)
+            mod = load_python_module(config_path)
+            _dict_merge(mod.CONFIG, saml2_config_dict)
```

Comment:
So `mod.CONFIG` overwrites the values specified in the config file?

Reply:
that's the current behaviour. I think it would be odd to put config in both the config file and an external config_path (in fact, I think it would be odd to use a config_path at all, but it's been supported for ever so I'm reluctant to just rip it out).